src/backend/binaryen/wasm_expr_gen.ts

```diff
@@ -770,6 +770,17 @@ export class WASMExpressionGen {
             !FunctionalFuncs.treatAsAny(leftValueType.kind) &&
             FunctionalFuncs.treatAsAny(rightValueType.kind)
         ) {
+            // the positions of the l-operand and r-operand are interchanged, and binary operators need to be reversed.
+            if (opKind == ts.SyntaxKind.LessThanEqualsToken) {
+                opKind = ts.SyntaxKind.GreaterThanEqualsToken;
```

Comment:
I think reverse the `opKind` here is very dangerous, we can not ensure the `opKind` is not used in some complex situation.

Reply:
this is exactly what you said